digitaltwin_client/src/dt_client_core.c

```diff
@@ -799,6 +799,58 @@ static DIGITALTWIN_CLIENT_RESULT SendInterfacesToRegisterMessage(DT_CLIENT_CORE*
     return result;
 }
 
+// VerifyInterfaceInstancesUnique makes sure that the interfaceInstanceName of each handle is unique.  E.G. one connection cannot have
+// to instances both named "frontCamera".  We check here, instead of solely relying on server policy, because if this fails on server
+// side for MQTT then the connection will be dropped and it will be hard for application developer to debug.
+// Duplicate interfaces ARE allowed - e.g. app can have two "urn:contoso:camera:1" interfaces as long as instances are different.
+static DIGITALTWIN_CLIENT_RESULT VerifyInterfaceInstancesUnique(DIGITALTWIN_INTERFACE_CLIENT_HANDLE* dtInterfaces, unsigned int numDTInterfaces)
+{
+    DIGITALTWIN_CLIENT_RESULT result = DIGITALTWIN_CLIENT_ERROR;
+    unsigned int i;
+    unsigned int j;
+
+    for (i = 0; i < numDTInterfaces; i++)
+    {
+        const char* interfaceInstanceName1 = DT_InterfaceClient_GetInterfaceInstanceName(dtInterfaces[i]);
+        if (interfaceInstanceName1 == NULL)
+        {   
+            LogError("DT_InterfaceClient_GetInterfaceInstanceName failed on element %d in handle list", i);
+            result = DIGITALTWIN_CLIENT_ERROR;
+            break;
+        }
+
+        for (j = i+1; j < numDTInterfaces; j++)
+        {
+            const char* interfaceInstanceName2 = DT_InterfaceClient_GetInterfaceInstanceName(dtInterfaces[j]);
+            if (interfaceInstanceName2 == NULL)
+            {   
+                LogError("DT_InterfaceClient_GetInterfaceInstanceName failed on element %d in handle list", j);
+                result = DIGITALTWIN_CLIENT_ERROR;
+                break;
+            }
+
+            if (strcmp(interfaceInstanceName1, interfaceInstanceName2) == 0)
```

Comment:
>strcmp [](start = 16, length = 6)

Please double check server case-sensitivity standard for instance names. Worth adding a comment here to specify server protocols. This is one area I miss our requirements comments :)

Reply:
I double checked before coding this :).  But agree a comment here is nice.

---
In reply to: [322953947](https://github.com/Azure/azure-iot-sdk-c/pull/1169#discussion_r322953947) [](ancestors = 322953947)